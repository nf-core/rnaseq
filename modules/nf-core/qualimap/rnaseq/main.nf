process QUALIMAP_RNASEQ {
    tag "$meta.id"
    label 'process_medium'

    conda "bioconda::qualimap=2.2.2d"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/qualimap:2.2.2d--1' :
        'quay.io/biocontainers/qualimap:2.2.2d--1' }"

    input:
    tuple val(meta), path(bam)
    path  gtf

    output:
    tuple val(meta), path("${prefix}"), emit: results
    path  "versions.yml"              , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args   ?: ''
    prefix   = task.ext.prefix ?: "${meta.id}"
    def paired_end = meta.single_end ? '' : '-pe'
    def memory = (task.memory.mega*0.8).intValue() + 'M'

    def strandedness = 'non-strand-specific'
    if (meta.strandedness == 'forward') {
        strandedness = 'strand-specific-forward'
    } else if (meta.strandedness == 'reverse') {
        strandedness = 'strand-specific-reverse'
    }
    """
    unset DISPLAY
    mkdir -p tmp
    export _JAVA_OPTIONS=-Djava.io.tmpdir=./tmp
    qualimap \\
        --java-mem-size=$memory \\
        rnaseq \\
        $args \\
        -bam $bam \\
        -gtf $gtf \\
        -p $strandedness \\
        $paired_end \\
        -outdir $prefix

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        qualimap: \$(echo \$(qualimap 2>&1) | sed 's/^.*QualiMap v.//; s/Built.*\$//')
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir ${prefix}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        qualimap: \$(echo \$(qualimap 2>&1) | sed 's/^.*QualiMap v.//; s/Built.*\$//')
    END_VERSIONS
    """
}
