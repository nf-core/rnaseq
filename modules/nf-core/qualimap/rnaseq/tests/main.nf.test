nextflow_process {

    name "Test Process QUALIMAP_RNASEQ"
    script "../main.nf"
    process "QUALIMAP_RNASEQ"
    tag "modules"
    tag "modules_nfcore"
    tag "qualimap"
    tag "qualimap/rnaseq"

    test("homo_sapiens [bam]") {

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true)
                ])
                input[1] = Channel.of([
                    [ id:'test_fasta_gtf' ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert path("${process.out.results[0][1]}/qualimapReport.html").exists() },
                { assert snapshot(path("${process.out.results[0][1]}/rnaseq_qc_results.txt")).match("rnaseq_qc_results") },
                { assert snapshot(process.out.versions).match("versions") }
            )
        }
    }
}
