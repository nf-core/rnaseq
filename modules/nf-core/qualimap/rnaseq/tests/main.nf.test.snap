{
    "rnaseq_qc_results": {
        "content": [
            "rnaseq_qc_results.txt:md5,b77878cac45beaa79a892af54aad2da3"
        ],
        "timestamp": "2024-01-19T12:07:55.509784"
    },
    "versions": {
        "content": [
            [
                "versions.yml:md5,a6868ab89f7d31361a7791db2dea98e7"
            ]
        ],
        "timestamp": "2024-01-19T12:07:55.512116"
    }
}